### PR TITLE
Jsonrpc talk endpoint support

### PR DESCRIPTION
**Pull request by njgheorghita**
## What was wrong?
Add support for the following jsonrpc endpoints
- `discv5_talk`
- `discv5_sendTalkRequest`
- `discv5_sendTalkResponse`

## How was it fixed?
Added support for the endpoints. 

### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [x] Clean up commit history

[//]: # (For important changes that should go into the release notes please add a newsfragment file as explained here: https://github.com/ethereum/ddht/blob/master/newsfragments/README.md)

[//]: # (See: https://ddht.readthedocs.io/en/latest/contributing.html#pull-requests)
- [x] Add entry to the [release notes](https://github.com/ethereum/ddht/blob/master/newsfragments/README.md)

#### Cute Animal Picture
image

### CODE REVIEW

File: ddht/tools/web3.py [hunk 1]
```diff
@@ -343,13 +362,36 @@ class DiscoveryV5Module(ModuleV2):  # type: ignore
         result_formatters=lambda method: find_nodes_response_formatter,
         mungers=[find_nodes_munger],
     )
-    send_find_nodes = Method(
+    send_find_nodes: Method[
```
Comment (njgheorghita) [inline ddht/tools/web3.py:365]: Any idea why mypy was missing these types? They still aren't being correctly validated btw...

**Comment by njgheorghita** [comment]
@pipermerriam Most of the methods that the jsonrpc endpoints point towards accept an optional `endpoint` or `request_id`. Currently, when using jsonrpc, these args default to `None`. Is it worthwhile enhancing the jsonrpc endpoints to accept these optional parameters?

File: ddht/tools/web3.py (hunk 1 above)
Comment (pipermerriam) [reply ddht/tools/web3.py:365]: If you're saying that these type hints aren't actually being validated (i.e. that you could changed one of them arbitrarily and mypy wouldn't complain) then I think that's something we should investigate (probably separately from this PR).  I'd recommend starting by using some `reveal_type` statements to figure out what mypy thinks these types are.

My guess is that web3.py might be leaking `Any` which is typically where this type of deficiency shows up.

**Review by pipermerriam — APPROVED** [review]

**Comment by pipermerriam** [comment]
> @pipermerriam Most of the methods that the jsonrpc endpoints point towards accept an optional `endpoint` or `request_id`. Currently, when using jsonrpc, these args default to `None`. Is it worthwhile enhancing the jsonrpc endpoints to accept these optional parameters?

Not crutial.  We can add it later.

**Pull request merged**